frontend/src/components/Blocker.tsx

```diff
@@ -0,0 +1,47 @@
+import Modal from "antd/lib/modal/Modal";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+
+interface BlockerProps {
+    children: ReactElement,
+    afterSeconds: number,
+    message: string
+}
+
+const Blocker: FunctionComponent<BlockerProps> = ({ afterSeconds, children, message }) : ReactElement => {
+    const [lastTap] = useState(new Date());
+    const [currentTime, setCurrentTime] = useState(new Date());
+    const [hasShown, setHasShown] = useState(false);
+    let onClick = children.props.onClick;
+    let newOnClick = () => {
+        if (onClick) {
+            onClick()
+        }
+        setHasShown(true);
+    }
+    useEffect(() => {
+        let interval = window.setInterval(() => {
+            setCurrentTime(new Date())
+        }, 1000)
+        return () => clearInterval(interval);
+    }, [])
+    const timeSince = currentTime.getTime() - lastTap.getTime();
+    let shouldShowBlocker = timeSince >= afterSeconds * 1000;
+
```

Comment:
Is it possible to make it repeatable? So basically like onOk, if the nextButton is still not clicked, and after the same amount of time, show it again

Reply:
I will work on this, but it _should_ be possible